internal/resources/metal/project_ssh_key/datasource.go

```diff
@@ -65,21 +60,18 @@ func (r *DataSource) Read(
 		return
 	}
 
+	keys := keysList.GetSshKeys()
 	for i := range keys {
-		// use the first match for searches
-		if search != "" {
-			key = keys[i]
-			break
-		}
+		key = keys[i]
```

Comment:
This is a major change in logic:  the old code took the first match, but this one will take the last.  The old code should remain in place.